Code/GraphMol/FileParsers/MolSGroupWriting.cpp

```diff
@@ -644,44 +675,60 @@ const std::string GetV3000MolFileSGroupLines(const unsigned int idx,
-  os << FormatV3000FieldDataBlock(sgroup);
-  os << FormatV3000StringPropertyBlock("CLASS", sgroup);
-  os << FormatV3000AttachPointBlock(sgroup.getAttachPoints());
-  os << FormatV3000StringPropertyBlock("BRKTYP", sgroup);
-  os << FormatV3000StringPropertyBlock("SEQID", sgroup);
-
-  std::string sGroupBlock = os.str();
-  unsigned int length = sGroupBlock.size();
-  os.str("");
-
-  unsigned int start = 0;
-  while (length - start > 73) {
-    os << "M  V30 " << sGroupBlock.substr(start, 72) << '-' << std::endl;
-    start += 72;
-  }
-  os << "M  V30 " << sGroupBlock.substr(start, 73) << std::endl;
-
-  return os.str();
+  addBlockToSGroupString(BuildV3000BondsBlock(sgroup), currLine, os);
+  addBlockToSGroupString(
+      BuildV3000IdxVectorDataBlock("PATOMS", sgroup.getParentAtoms()), currLine,
+      os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("SUBTYPE", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("MULT", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("CONNECT", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000ParentBlock(sgroup), currLine, os);
+  addBlockToSGroupString(FormatV3000CompNoBlock(sgroup), currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("LABEL", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000BracketBlock(sgroup.getBrackets()),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("ESTATE", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000CStateBlock(sgroup), currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("FIELDNAME", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("FIELDINFO", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("FIELDDISP", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("QUERYTYPE", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("QUERYOP", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000FieldDataBlock(sgroup), currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("CLASS", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000AttachPointBlock(sgroup.getAttachPoints()),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("BRKTYP", sgroup),
+                         currLine, os);
+  addBlockToSGroupString(FormatV3000StringPropertyBlock("SEQID", sgroup),
+                         currLine, os);
+  std::string res;
+  if (!currLine.empty() && currLine != "M  V30") {
+    os << currLine << std::endl;
```

Comment:
I would stick with "\n" here for consistency.

Reply:
Unfortunately we've currently got a mix of `"\n"` and `std::endl` throughout this file, MolFileWriter.cpp, and other places.
Cleaning that up and probably using `"\n"` everywhere there's not a clear need to flush the stream would be a good future cleanup PR.